merge 2019.06 to master

kkrebs opened the pull request.

Commit:
MCR-2057 Field list in expert search is sometimes missing

kkrebs committed:
MCR-2041 replace glyphicons with fontawesome v5 ...

... and fixed pagination in solr response and acl editor

sebhofmann committed:
MCR-2078 cron prevent update if date didnt change

MyCoRe-Travis commented:
Something gone wrong on f04257b2d5a91e512a11f30fc83967aad6a33e9a! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/4738-pull_request-832

Commit:
MCR-2077 framed epub viewer now loads

MyCoRe-Travis commented:
Something gone wrong on e1a71ad2aa353901b1c12c8c73a9f289cbb56721! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/4742-pull_request-832

OPNA2608 committed:
MCR-2084 Configurable IView2 preview image size

MCRTileCombineServlet permits several image sizes, but the `iview2.getImageElement` template
hardcodes THUMBNAIL. In a use case where `derivateView` is intended to be bigger than just
a small preview, this does not look good.

This adds the param `imageSize`, to make the size configurable when calling `derivateView`.
The impactful values are (as documented on the IView2 frontend) THUMBNAIL, MIN, MID and MAX,
any other input results in the THUMBNAIL size.

This change is backwards-compatible, the default value of `imageSize` is THUMBNAIL.

bt1cu committed:
MCR-1792 update to mods version 3.7

* modify MODS-schema from LOC for using native xlink schema

MyCoRe-Travis commented:
Something gone wrong on 9e619519ed175f2de32171e5149dfd77d3425089! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/4769-pull_request-832

sebhofmann committed:
MCR-1792 recreate tei xsd files

Commit:
MCR-2081 epub viewer sometimes jumps to wrong chapter

MyCoRe-Travis commented:
Something gone wrong on c76d1d03480c8d36200db41f2008678a53e4e635! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/4786-pull_request-832

fluetze committed:
MCR-2088 Fixed NPE in case WEB-INF/classes/xsl/ is not used

sebhofmann committed:
MCR-2041 repaired upload gui layout

TiloNN committed:
MCR-2041 replaces bs3 validation css classes

* add .mcr-invalid and .mcr-valid css classes as workaround to acl editor

sebhofmann committed:
MCR-2091 epub viewer does not work in ie

MyCoRe-Travis commented:
Something gone wrong on e378565f9f0c869c86775b3b07674bde4877c8be! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/4811-pull_request-832

sebhofmann committed:
MCR-2041 fix some display issues

sebhofmann committed:
MCR-1978 use static initializer to init static variables

kkrebs committed:
add messages properties for hard coded values

Commit:
MCR-2043 override static xml files

rsteph-de committed:
MCR-2099 update derivate migration command

* ignore object type at service flag migration
* add derivate data to solr and to db
* fix wrong xpath
* fix command syntax
* add servstate of derivate to link table
* change mainDoc to maindoc

rsteph-de committed:
MCR2102 fix "fixed attributes" in MODS schema

Commit:
MCR-2082 - improve mods crossref xsl

kkrebs committed:
MCR-2041 - fix bootstrap migration errors

* fixed wrong offset classes and deletion message
* fix maximize button and fa to fas
* fixed overview toggle icon
* fix icon class and field name

rsteph-de committed:
MCR-2104 - make MCRUser$Password serializable

eckstueck committed:
MCR-2090 add support for IIIF to viewer (#843)

* MCR-2090 add support for IIIF to viewer
first version of IIIF support
you can use IIIF by adding iiif to the path (viewer/iiif/{derivateID}/)
removed Alto support with iiif for now, will be added later
merged classes that shared code
fixed some codacy issues
fixed some more codacy issues

kkrebs committed:
MCR-2099 - fix name of derivate search field maindoc (#859)

... it cause problems with usage of mycoreobject-dynamicfields.xsl and new derobject xml

MyCoRe-Travis commented:
Something gone wrong on b9461a5f86426467b09051b7ff76a4d2b8654811! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/4934-pull_request-832

Possommi committed:
MCR-2101 - MCRJobQuere should have MCRJob getJobs public (#856)

* MCR-2101 added public List<MCRJob> getJobs(Map<String, String> params) and private List<MCRJob> getJobs(Class<? extends MCRJobAction> action, Map<String, String> params) to org.mycore.services.queuedjob.MCRJobQueue

* MCR-2101 MCRJobQueue getJob(s) now works with CriteriaQuery

* change travis to xenial

* use openjdk8 in travis

rsteph-de committed:
MCR-2055 - fix incorrect stream handling

* add toString() to improve debug messages
* close directory streams (often created by Files.list())

sebhofmann committed:
Merge branch '2018.06.0.x' into 2019.06.x

* 2018.06.0.x:
  MCR-2078 cron prevent update if date didnt change
  MCR-2101 - MCRJobQuere should have MCRJob getJobs public (#856)
  add messages properties for hard coded values
